### PR TITLE
MNTOR-1915 - use APP_ENV to differentiate environments per NextJS recommendation

### PR DESCRIPTION
**Pull request by rhelmer**
# References: 
Jira: MNTOR-1915

# Description

Use `NODE_ENV` and `APP_ENV` per https://nextjs.org/docs/messages/non-standard-node-env

# How to test

Everything should function as before, but we should take extra care to test that Sentry handles this correctly.

# Checklist (Definition of Done)
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

**Comment by mansaj** [comment]
There's also `sentry.client.config.ts` where we might want to modify line 15

File: src/app.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ import { noSearchEngineIndex } from './middleware/noSearchEngineIndex.js'
 import { RateLimitError } from './utils/error.js'
 
 const app = express()
-const isDev = AppConstants.NODE_ENV === 'dev'
+const isDev = AppConstants.NODE_ENV === 'development'
 
 // init sentry
 Sentry.init({
```
Comment (mansaj) [inline src/app.js:30]: For sentry, we might have to use `APP_ENV` to differencitate

File: src/appConstants.js [hunk 2]
```diff
@@ -65,8 +66,7 @@ const optionalEnvVars = [
 /** @type {Record<string, string>} */
 const AppConstants = { }
 
-// @ts-ignore TS thinks NODE_ENV can't be "heroku". Shut up TS, you're drunk!
-if (!process.env.SERVER_URL && (process.env.NODE_ENV) === 'heroku') {
+if (!process.env.SERVER_URL && (process.env.APP_ENV) === 'heroku') {
```
Comment (mansaj) [inline src/appConstants.js:69]: @Vinnl sorry we have to take out this comment!

**Review by mansaj — CHANGES_REQUESTED** [review]
Just a couple of things...
Now the real work begins - adding /changing env vars to all the environments

File: src/app.js (hunk 1 above)
Comment (rhelmer) [reply src/app.js:30]: OK I looked in docs, we can pass `environment` to the `Sentry.init` function and use `APP_ENV`.

**Comment by rhelmer** [comment]
@mansaj thanks! I found a few more places, and I think I see how we can use `APP_ENV` to set the Sentry environment. This means that all environments will either have `NODE_ENV="development"` (local, heroku) or `NODE_ENV="production"`. NextJS will use this as a signal to run optimizations (and ESBuild for the remaining Express code).

`APP_ENV` may be `local`, `heroku`, `stage`, or `production`. These will be used for the Sentry environment and anywhere else we need per-environment config.

**Review by mansaj — APPROVED** [review]

File: sentry.client.config.ts [hunk 3]
```diff
@@ -9,12 +9,11 @@
...
 
 Sentry.init({
+  environment: process.env.NEXT_PUBLIC_APP_ENV,
   dsn: process.env.NEXT_PUBLIC_SENTRY_DSN,
 
   // Adjust this value in production, or use tracesSampler for greater control
-  tracesSampleRate: ["development", "heroku"].includes(process.env.NODE_ENV)
-    ? 1.0
-    : 0.1,
+  tracesSampleRate: ["development"].includes(process.env.NODE_ENV) ? 1.0 : 0.1,
```
Comment (pdehaan) [inline sentry.client.config.ts:16]: very ignorant and lazy Q: is "NODE_ENV" special? Why wouldn't this need a "NEXT_PUBLIC_" prefix too?

**Comment by pdehaan** [comment]
@rhelmer: Is this matrix roughly accurate?

ENV | `NODE_ENV` | `APP_ENV`
:----|:----|:----
local | "development" | "local"
heroku/dev | "development" | "heroku"
ci | "tests" | "local" ❓ 
stage | "production" | "stage"
production | "production" | "production"

**Pull request merged**